upload/upload_test.go

```diff
@@ -163,6 +166,82 @@ func TestUnitUpload(t *testing.T) {
 
 		})
 
+		Convey("test upload successfully uploads with only one chunk and a valid vault client", func() {
```

Comment:
is it reasonable to write tests for if vault isn't configured or is configured wrongly? or any other failure scenarios? ('too much mocking to be valuable' is totally an acceptable reason not to)

Reply:
"too much mocking" is my answer - it would take many, many tests to test all error scenarios just to test for a internal server error status code.